hack/cmd_util.sh

```diff
@@ -234,3 +268,120 @@ function os::cmd::internal::assemble_causes() {
 	list=$(printf '; %s' "${causes[@]}")
 	echo "${list:2}"
 }
+
+
+# os::cmd::internal::run_until_exit_code runs the provided command until the exit code test given 
+# succeeds or the timeout given runs out. Output from the command to be tested is suppressed unless
+# either `VERBOSE=1` or the test fails. This function bypasses any error exiting settings or traps
+# set by upstream callers by masking the return code of the command with the return code of setting
+# the result variable on failure.
+function os::cmd::internal::run_until_exit_code() {
+	cmd=$1
+	cmd_eval_func=${2}
+	duration=$3
+
+	os::cmd::internal::init_tempdir
+
+	echo "Waiting on ${cmd}..."
+	
+	deadline=$(( $(date +%s000) + $duration ))
+	while [ $(date +%s000) -lt $deadline ]; do	
+		cmd_result=$( os::cmd::internal::run_collecting_output "${cmd}"; echo $? )
+		cmd_succeeded=$( ${cmd_eval_func} "${cmd_result}"; echo $? )
+		if (( cmd_succeeded )); then
+			break
+		fi
```

Comment:
Ah, interesting. `tryuntil` does, `wait_for_command` sleeps 200ms. I can see why the latter is useful. I'll update these to do that as well.